compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1609,44 +1609,64 @@ pub type PlaceholderType = Placeholder<BoundVar>;
 pub type PlaceholderConst = Placeholder<BoundVar>;
 
 /// A `DefId` which is potentially bundled with its corresponding generic parameter
-/// in case `did` is a const argument.
-///
-/// This is used to prevent cycle errors during typeck
-/// as `type_of(const_arg)` depends on `typeck(owning_body)`
-/// which once again requires the type of its generic arguments.
-///
-/// Luckily we only need to deal with const arguments once we
-/// know their corresponding parameters. We (ab)use this by
-/// calling `type_of(param_did)` for these arguments.
+/// in case `did` is a const argument. Used to prevent cycle errors during `typeck`.
```

Comment:
"its corresponding generic parameter" comes very much out of the blue here... what bout?
```
/// A `DefId, which (if it is a const *argument*) is potentially bundled with a generic *parameter* whose
/// type determines the type of the argument. Avoids queries on the argument, which prevents
/// cycle errors during `typeck`.
```